metrics/open-telemetry-backend/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryBackend.scala

```diff
@@ -0,0 +1,79 @@
+package sttp.client3.opentelemetry
+
+import io.opentelemetry.api.common.{AttributeKey, Attributes}
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator
+import io.opentelemetry.api.trace.{Span, SpanKind, StatusCode, Tracer}
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.propagation.{TextMapPropagator, TextMapSetter}
+import sttp.capabilities.Effect
+import sttp.client3._
+import sttp.monad.MonadError
+import sttp.monad.syntax._
+
+import scala.collection.mutable
+
+private class OpenTelemetryBackend[F[_], P](
+    delegate: SttpBackend[F, P],
+    tracer: Tracer
+) extends SttpBackend[F, P] {
+
+  private implicit val _monad: MonadError[F] = responseMonad
+  type PE = P with Effect[F]
+
+  def send[T, R >: PE](request: Request[T, R]): F[Response[T]] = {
```

Comment:
the impl looks quite different from the zio one, but I suspect that's beacuse the zio library was doing some of the work that happens here? I'm not familiar with other opentelemetry integrations, if that's what is usually done, fine :)

Reply:
I used example from `io.opentelemetry.api.trace.Tracer` class with automatic context propagation it seemed reasonable